### PR TITLE
Do not canonicalize the file names when getting absolute paths during nodenext resolution

### LINKED ISSUE #50544: `forceConsistentCasingInFileNames: true` and `moduleResolution: Node16` break type reference directives on macOS if there are capital letters in the path
# Bug Report

### 🔎 Search Terms

`forceConsistentCasingInFileNames`, `moduleResolution`, `Node16`, `NodeNext`, type reference, "differs from already included file name", "only in casing". Relevant issues (all resolved and predating `Node16`): https://github.com/microsoft/TypeScript/issues/16875, https://github.com/microsoft/TypeScript/issues/20003, https://github.com/microsoft/TypeScript/issues/17617, https://github.com/microsoft/TypeScript/issues/9824

### 🕗 Version & Regression Information

~This is the behavior in every version I tried, and I haven't found any issues or FAQ entries about this.~ Tested only on 4.8.2, sorry. Looks like this might be recent regression after all.

### ⏯ Playground Link / 💻 Code

I can't reproduce this in neither the playground nor the workbench nor even Linux. The problem appears only on macOS. Here's a repro: https://github.com/alecmev/typescript-type-reference-lowercase-repro Just `npm i` and `npx tsc`.

### 🙁 Actual behavior

```bash
index.ts:2:26 - error TS1149: File name '/users/foo/repro/node_modules/@types/react/index.d.ts' differs from already included file name '/Users/Foo/repro/node_modules/@types/react/index.d.ts' only in casing.
  The file is in the program because:
    Type library referenced via 'react' from file '/Users/Foo/repro/node_modules/@babylonjs/inspector/babylon.inspector.module.d.ts'
    Imported via "react" from file '/Users/Foo/repro/node_modules/@babylonjs/inspector/babylon.inspector.module.d.ts' with packageId '@types/react/index.d.ts@18.0.17'
```

Note `/users/foo` vs. `/Users/Foo`.

### 🙂 Expected behavior

This is valid code and it works on Linux and/or without `moduleResolution: Node16`. An unnecessary `toLowerCase` happens somewhere, but I can't really debug this further since I use Linux and had to use my colleague's computer to create the repro. The title of the issue is only a guess, I don't know the root of the issue.

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #50544

### CODE REVIEW

File: src/testRunner/unittests/tscWatch/forceConsistentCasingInFileNames.ts [hunk 1]
```diff
@@ -262,5 +262,49 @@ a;b;
...
+                    name: "yargs",
+                    version: "17.0.12",
+                    exports: {
+                        ".": {
+                            types: {
+                                import: "./index.d.mts",
+                                default: "./index.d.ts"
+                            }
+                        },
+                        // "./helpers": {
```
Comment (weswigham) [inline src/testRunner/unittests/tscWatch/forceConsistentCasingInFileNames.ts:284]: ?

**Review by weswigham — APPROVED** [review]
Looks ok, aside from some commented-out code in the test.

I do worry that this means we'll now only match exports paths with exact case matches, even on case-insensitive systems, but weather that's OK or not I guess depends on node's behavior?

**Review by ProKashif — APPROVED** [review]

File: src/testRunner/unittests/tscWatch/forceConsistentCasingInFileNames.ts (hunk 1 above)
Comment (sheetalkamat) [reply src/testRunner/unittests/tscWatch/forceConsistentCasingInFileNames.ts:284]: Sorry i was copy pasting and forgot to remove this.. will do that

**Comment by sheetalkamat** [comment]
> I do worry that this means we'll now only match exports paths with exact case matches, even on case-insensitive systems, but weather that's OK or not I guess depends on node's behavior?

I will lookup some cases and try to get that as well since we have methods that handle casing to check if paths match that way using case without having to convert the path to canonical form

**Comment by sheetalkamat** [comment]
@weswigham fixed other references where absolute path was used to check canonically

**Comment by sheetalkamat** [comment]
@typescript-bot cherry-pick this to release-4.8

**Review by weswigham — APPROVED** [review]

**Comment by alecmev** [comment]
Thank you!

**Comment by JounQin** [comment]
`4.8.3` and `4.9.0-dev.20220912` both do not work to me.

https://github.com/un-ts/lib-boilerplate/runs/8301954091?check_suite_focus=true#step:6:37

**Comment by sheetalkamat** [comment]
Looking into this

**Pull request merged**